pkg/apis/awsprovider/v1alpha1/awsprovider.go

```diff
@@ -0,0 +1,39 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AWSProviderSpec is the top level specification for the AWS Karpenter Provider.
+// This will contain configuration necessary to launch instances in AWS.
+type AWSProviderSpec struct {
+	// UserData to be applied to the provisioned nodes.
+	// It must be in the appropriate format based on the AMIFamily in use. Karpenter will merge certain fields into
+	// this UserData to ensure nodes are being provisioned with the correct configuration.
+	// +optional
+	UserData *string `json:"userData,omitempty"`
```

Comment:
I was tempted to inline `v1alpha1.AWS` in here from `pkg/cloudprovider/aws/apis/v1alpha1` but I decided not since then we'd end up complicating a lot of our logic trying to read values from either of the two places and uniting the defaulting / validating webhook logic.

 I figured it's easier to always read `UserData` from this CRD and all the other configuration values remain within the Provisioner CRD until we do a clean switch as part of an API upgrade.

Reply:
+1!